enterprise/internal/campaigns/resolvers/campaign_spec.go

```diff
@@ -200,3 +203,51 @@ func (r *campaignSpecResolver) AppliesToCampaign(ctx context.Context) (graphqlba
 		Campaign:    campaign,
 	}, nil
 }
+
+func (r *campaignSpecResolver) ViewerCampaignsCodeHosts(ctx context.Context, args *graphqlbackend.ListViewerCampaignsCodeHostsArgs) (graphqlbackend.CampaignsCodeHostConnectionResolver, error) {
+	actor := actor.FromContext(ctx)
+	if !actor.IsAuthenticated() {
+		return nil, backend.ErrNotAuthenticated
+	}
+
+	// Short path for site-admins when OnlyWithoutCredential is true: It will always be an empty list.
+	if args.OnlyWithoutCredential {
+		if authErr := backend.CheckCurrentUserIsSiteAdmin(ctx); authErr == nil {
+			// For site-admins never return anything
+			return &campaignsCodeHostConnectionResolver{
+				store: r.store,
+				customCompute: func(ctx context.Context) (all []*campaigns.CodeHost, page []*campaigns.CodeHost, credsByIDType map[idType]*db.UserCredential, err error) {
+					return []*campaigns.CodeHost{}, []*campaigns.CodeHost{}, nil, nil
+				},
+			}, nil
```

Comment:
I might be missing something, but it seems like this is the only place where we use a `customCompute` function and all we do is to return an empty list.

If that's the case, how about we define this somewhere:

```go
type emptyCampaignsCodeHostConnectionResolver struct{}

var _ graphqlbackend.CampaignsCodeHostConnectionResolver = &emptyCampaignsCodeHostConnectionResolver{}

func (c *emptyCampaignsCodeHostConnectionResolver) TotalCount(ctx context.Context) (int32, error) {
	return int32(0), nil
}

func (c *emptyCampaignsCodeHostConnectionResolver) PageInfo(ctx context.Context) (*graphqlutil.PageInfo, error) {
	return graphqlutil.HasNextPage(false), nil
}

func (c emptyCampaignsCodeHostConnectionResolver) Nodes(ctx context.Context) ([]graphqlbackend.CampaignsCodeHostResolver, error) {
	return []graphqlbackend.CampaignsCodeHostResolver{}, nil
}
```

and then simply return that here:

```suggestion
			return &emptyCampaignsCodeHostConnectionResolver{}, nil
```